src/haz3lcore/projectors/implementations/CheckboxProj.re

```diff
@@ -0,0 +1,68 @@
+open Util;
+open ProjectorBase;
+open Virtual_dom.Vdom;
+
+module M: Projector = {
+  [@deriving (show({with_path: false}), sexp, yojson)]
+  type model = unit;
+  [@deriving (show({with_path: false}), sexp, yojson)]
+  type action = unit;
+
+  let bool_of = (any: Any.t): option(bool) =>
+    switch (any) {
+    | Exp({term: Bool(b), _}) => Some(b)
+    | _ => None
+    };
+
+  let init = (any: Term.Any.t) =>
+    switch (bool_of(any)) {
+    | Some(_) => Some()
+    | None => None
+    };
+
+  let get = (info: info): bool =>
+    switch (
+      info.syntax |> info.utility.seg_to_term |> OptUtil.and_then(bool_of)
+    ) {
+    | Some(b) => b
+    | None => failwith("Checkbox: Get: not boolean literal")
+    };
+
+  let toggle = (info): Base.segment =>
+    switch (
+      info.utility.lift_syntax(
+        fun
+        | Exp({term: Bool(b), _} as t) => Exp({...t, term: Bool(!b)})
+        | _ => failwith("Checkbox: Toggle: not boolean literal"),
+        info.syntax,
+      )
```

Comment:
I know we have some functions that operate on the segment structure as it were an expression now but I wonder if this is a sign that the checkbox projector should have a boolean model and just set the segment to whatever the model is. Then toggle the model.

Reply:
The projectors PR matt and I are working on now will hopefully eliminate the distinction between a projector model and the underlying syntax in favor of a model with a make_term method. Until then I think this is necessarily awkward. But it's not obvious to me what adding a dual source of truth would add at the moment

Reply:
Yeah, we can probably wait for a more general fix. I still think it's less of a second source of truth and just making the model the source of truth in this example.

Reply:
but regardless of what we put in the model there's still going to be a (syntactic) boolean in the underlying syntax

Reply:
Agreed. It just seems simpler to me to not have any error handling outside of the model initialization instead of on every transformation.